README.md
# LTspice_automation

Changelog.txt
01/25/23 LT8315 Changed diodes from obsolete parts to currently supported parts.
01/25/23 LT4322  new model
01/24/23 MAX16543  new model
01/24/23 MAX16545B  new model
01/24/23 MAX16545C  new model
01/24/23 MAX16550  new model
01/24/23 MAX17526A  new model
01/24/23 MAX17612A  new model
01/24/23 MAX17612B  new model
01/24/23 MAX17612C  new model
01/24/23 MAX17613A  new model
01/24/23 MAX17613B  new model
01/24/23 MAX17613C  new model
01/24/23 MAX20313  new model
01/24/23 MAX20314  new model
01/24/23 MAX20315  new model
01/24/23 MAX20316  new model
01/23/23 LTC7150S Corrected CLKOUT output signal; Updated pin assignments
01/23/23 LTC7890 corrected clock synchronization function of PLL pin; corrected Mode functions
01/23/23 LTC7891 corrected clock synchronization function of PLL pin
01/23/23 MAX40023 Reduced overload recovery time.
01/23/23 AD8336  new model
01/22/23 LTC7103 Added Boost pin.
01/17/23 LT3154-1 removed
01/17/23 W5AO-LZMZ-5K8L removed as requested
01/17/23 LTC1261CS8 Fix stability and ripple issue
01/17/23 LTC7891 corrected Mode functions
01/16/23 LT8643S Fix Negative Vout config, BIAS, Vin1, and Vin2 Currents,offset on FB Pin.
01/12/23 LT8650S Fixed CLKout signal when in parallel operation
01/12/23 LT3073  new model
01/09/23 Added capacitors from Vishay in standard capacitors library
01/06/23 LTM4660 Fixed the PLL function and improved convergence.
01/05/23 LTC7890 corrected function of DTCx Pins Connected with a Resistor to GND
01/05/23 LTC7891 corrected function of DTCx Pins Connected with a Resistor to GND
01/05/23 ADA4098-2 Corrected symbol reference to netlist.
01/05/23 MAX41400 Improved convergence performance.
01/03/23 LT3097  new model
12/21/22 LTC1261CS8 Control circuit changes to reduce ripple to better match silicon.
12/15/22 Updated standard capacitor library to use vendor Chemi-Con over United Chemi-con (local standard.cap.bak will be added locally)
12/14/22 LTC3899 Changed symbol for application circuit.
12/14/22 LTC7871 Updated symbol.
12/13/22 LT8333 Update Slope compensation profile to match LTpowerCAD
12/13/22 LT8334 Update Slope compensation profile to match LTpowerCAD
12/13/22 LTspice 17.0.36 with minor bug fixes
12/12/22 MAX17561  new model
12/12/22 MAX17562  new model
12/12/22 MAX17563  new model
12/08/22 LTC6563 Revised model for higher bandwidth
12/07/22 LTC4286  new model
12/06/22 ADM1272 Fast response to severe overcurrent
12/05/22 MAX40023 Model updated to new architecture with accuracy and convergence improvements.
12/02/22 MAX17608  new model
12/02/22 MAX17609  new model
12/02/22 MAX17610  new model
11/28/22 ADM1272 ESTART/EFAULT pin current, updated symbol pin assignment
11/28/22 LTC7890 corrected dead-time control
11/28/22 LTC7891 corrected dead-time control
11/28/22 LTC3899 Added logic so part would go into shutdown.
11/28/22 LTC7871 Modified circuit so V5 and DRVCC would be disabled when all run pins pulled low.
11/22/22 ADL5304  new model
11/22/22 ADL5304  new model
11/21/22 AD4030-24 Improved convergence
11/21/22 ADR1000 Optimized model to improve convergence. Updated example circuit. 
11/21/22 MAX41400 Corrected internal node references.
11/17/22 LTspice help file updated providing link to LTspice 17.1 beta 
11/17/22 LTM4657 Fix SS currrent issue
11/16/22 MAX41400 Model updated to new architecture with accuracy and convergence improvements.
11/16/22 LT1943 Corrected error with .include statement.
11/16/22 LTM4663 Corrected error with .include statement.
11/16/22 ADN8834 Corrected error with .include statement.
11/14/22 AD8422 Changed Quiescent Current
11/11/22 ADA4896 Updated symbol to reference new model.
11/11/22 ADA4896-2 Model updated to new architecture with accuracy and convergence improvements.
11/11/22 LT3030 Added VIN UVLO as a condition that disables the part.
11/11/22 LTM8022 Revised PGood logic
11/10/22 ADGM1144  new model
11/09/22 LTC2057HV Updated the symbol
11/09/22 LTC2057 Updated the symbol
11/09/22 LTC2058 Updated the symbol
11/09/22 LT8584 Simple Mode Operation; DIN pin and DCHRG pin functions
11/09/22 LTC3290  new model
11/08/22 LTC6090 Update symbol pin placement for V+ and V-
11/08/22 LTC6090-5 Update symbol pin placement for V+ and V-
10/25/22 ADG1206  new model
10/25/22 ADG1207  new model
10/20/22 LT8648S Inverted PGood logic 
10/19/22 LTC3103 minor changes in Burst Model Operation
10/18/22 LTC7891 Modified Oscillator and Slope Compensation; Fix steady state detection 
10/18/22 LTC7890 Fixed steady state detection
10/18/22 LTC3311S Update control paramater value
10/17/22 LTC3311-0.875 Update the slope compensation
10/17/22 LTC3311-0.85 Update the slope compensation
10/17/22 LTC3311 Update the slope compensation
10/17/22 LTC3311-1 Update the slope compensation
10/17/22 LTC3311S Fixed the slope comp profile for the entire frequency range
10/17/22 LTC3311-0.75 Update the slope compensation
10/17/22 LT3041  new model
10/13/22 LT8710 Decrease the value to BG minimum ON time.
10/13/22 LTC2387-16 Updated noise model 
10/11/22 AD8310 Corrected error with .include statement.
10/11/22 LT1336 Fix issue on V+ function and shutdown profile
10/08/22 LTC6373 Improved accuracy of small-signal response.
10/08/22 ADR1001  new model
10/07/22 LTC7890 Fixed 2-phase operation and Vprog function
10/07/22 LT8643S Correct the SS pin to Reference function
10/07/22 ADGM1001  new model
10/07/22 MAX15090B  new model
10/07/22 MAX15090C  new model
10/07/22 MAX15093  new model
10/07/22 MAX15091A  new model
10/07/22 MAX15093A  new model
10/07/22 MAX15091  new model
10/06/22 LTC7890 Fixed 2-phase operation and Vprog function
10/06/22 ADGM1001  new model
09/30/22 LTM4658 Improved Load Transient Response
09/30/22 ADG5462F Added decoupling capacitors to the application circuit.
09/29/22 LTM4661 Soft-start timing was corrected
09/28/22 LTC4380-2 Fix OV and OC response
09/28/22 LTC4380-4 Fix OV and OC response

Trial_2.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jun 15 23:56:11 2023

@author: Tathagata Sau
"""

import systemcode as m
import os
import matplotlib.pyplot as plt
import xlsxwriter
import numpy as np
import pandas as pd
import os
import subprocess
import ltspy3
from sklearn.svm import SVR
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import mean_squared_error
from sklearn import metrics
from scipy.stats import skew
from scipy.stats import kurtosis

def chplot(n1,n2):

    ltspice_exe_path = r'C:\Program Files\LTC\LTspiceXVII\XVIIx64.exe'
    netlist_path = r"C:\Users\Tathagata Sau\OneDrive\Documents\LTspiceXVII\tmp5.cir"
    subprocess.call([ltspice_exe_path, '-Run', netlist_path])
    path = r"C:\Users\Tathagata Sau\OneDrive\Documents\LTspiceXVII"
    filename = "tmp5"
    path += '\\tmp5.raw'
    sd = ltspy3.SimData(filename=path)  
    name = sd.variables  
    freq_complex = sd.values  
    a = sd.values[n1]  
    b = sd.values[n2]  
    for i in range(201):
        plt.plot(a[0], b[i]*(-1))
    plt.xlabel('VDD(v)')
    plt.ylabel('I_out(nA)')
    plt.title('powersupplysensitivity for ch')
    v = len(a[0])  
    for j in range(v):
        if round(a[0][j], 2) >= 1.30:
            x = j
            break
    l1 = []
    for i in range(201):
        l1.append(b[i][x]*(-1))  
    l2 = []
    for i in range(int(170e-9 / 0.1e-9), int(190e-9 / 0.1e-9)):
        wavelength = (i * 0.1e-9)
        l2.append(wavelength)
    plt.show()
    createexcel(l1,l2)
    return l1

def createexcel(l1, l2):

    user_input = input("Please give a name for the channel length excel file : ") 
    newDataframe = pd.DataFrame()
    newDataframe['Channel Length'] = l2[0:201]
    newDataframe['I(A)'] = l1[0:201]
    newDataframe.to_excel(user_input+".xlsx", index = False)

def custom_SVM(df1):

    df=df1
    X = df.iloc[:, 1].values
    y = df.iloc[:, 0].values

    from sklearn.preprocessing import StandardScaler
    sc_X=StandardScaler()  
    sc_y=StandardScaler()
    X=sc_X.fit_transform(X.reshape(-1,1))
    y=sc_y.fit_transform(y.reshape(-1,1))

    from sklearn.model_selection import train_test_split
    X_train, X_test, y_train, y_test =  train_test_split(X,y,test_size=0.2)

    from sklearn.svm import SVR
    regressor=SVR(C=0.1, gamma=0.01, kernel='rbf')
    regressor.fit(X_train.reshape(-1,1), y_train.reshape(-1,1))
    y_pred = regressor.predict(X_test.reshape(-1,1))
    y_pred = sc_y.inverse_transform(y_pred.reshape(-1,1))
    y_pred=y_pred.ravel()
    z = sc_y.inverse_transform(y_test)
    print(z.ravel())
    c = df['I_out(nA)']
    res = []
    for mem in z.ravel():
        for i, mem1 in enumerate(c):
            if mem == mem1:
                res.append(df['tox values'][i])
    df = pd.DataFrame({'Tox': res, 'Real Values': z.ravel()[0:len(res)], 'Predicted Values': y_pred[0:len(res)]})
    print(df.to_string())

    mean=sum(y_pred)/len(y_pred)
    s=skew(y_pred)
    k=kurtosis(y_pred)
    print("For Predicted Values : ")
    print("Mean value is: ",mean)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)

    print('Mean Absolute Error is : ')
    print(metrics.mean_absolute_error(y_test,y_pred))
    print('Mean Squared Error is : ')
    print(metrics.mean_squared_error(y_test,y_pred))
    print('Root Mean Squared Error is :')
    print(np.sqrt(metrics.mean_squared_error(y_test,y_pred)))

def plot(current,n,n1):

    plot2=plt.figure(2)
    plt.hist(current,bins=n1)
    plt.xlabel('I_out(nA)')
    plt.ylabel('Frequency')
    if n==1:
        plt.title('I_out(nA) for tox values')
    elif(n==2):
        plt.title('I_out(nA) for vth values')
    else:
        plt.title('I_out(nA) for both variation')
    plt.show() 

def pathmodification(txtfile):

    l=txtfile.split('\\')
    l.pop(len(l)-1)
    path='\\'.join(l)
    return path 
  
def toxxl(current,tox,filename,path,n):

    path1 = "E://PROECT_Main//LTspiceXVII//LTspiceXVII"
    dir_list = os.listdir(path1)
    if('tox_n_varition_Final.xlsx' in dir_list):
        print("Duplicate Found")
        file = 'tox_n_varition_Final.xlsx'
        location = "E:/PROECT_Main/LTspiceXVII/LTspiceXVII/"
        path1 = os.path.join(location, file)
        os.remove(path1)
    os.chdir(path)
    filename+=".xlsx"
    a=xlsxwriter.Workbook(filename)
    b=a.add_worksheet()
    b.write("A1","I_out(nA)")
    b.write("B1","tox values")
    for i in range(n):
        b.write(i+1,0,current[i])
        b.write(i+1,1,tox[i])
    a.close()

def vthxl(current,vtharray,filename,path,n):

    path1 = "C://Users//Tathagata Sau//OneDrive//Documents//LTspiceXVII"
    dir_list = os.listdir(path1)
    if('VTH_n_varition_Final.xlsx' in dir_list):
        print("Duplicate Found")
        file = 'VTH_n_varition_Final.xlsx'
        location = "C:/Users/Tathagata Sau/OneDrive/Documents/LTspiceXVII/"
        path1 = os.path.join(location, file)
        os.remove(path1)
    os.chdir(path)
    filename+=".xlsx"
    a=xlsxwriter.Workbook(filename)
    b=a.add_worksheet()
    b.write("A1","I_out(nA)")
    j=66
    for i in range(18):
        if i<9:
            h='vthen'+str(i+1)+'n'
            x=chr(j)+'1'
            b.write(x,h)
            j+=1 
        else:
            h='vthep'+str(i-8)+'p'
            x=chr(j)+'1'
            b.write(x,h)
            j+=1
    for i in range(n):
        b.write(i+1,0,current[i])
    k=1
    for arr in vtharray:
        for i in range(n):
            b.write(i+1,k,arr[i])
        k+=1
    a.close() 

def vthandtoxxl(current,vtharray,toxvalues,filename,path,n):

    path1 = "C://Users//Tathagata Sau//OneDrive//Documents//LTspiceXVII"
    dir_list = os.listdir(path1)
    if('VTH_P_varition_Final.xlsx' in dir_list):
        print("Duplicate Found")
        file = 'VTH_P_varition_Final.xlsx'
        location = "C:/Users/Tathagata Sau/OneDrive/Documents/LTspiceXVII/"
        path1 = os.path.join(location, file)
        os.remove(path1)
    os.chdir(path)
    filename+=".xlsx"
    a=xlsxwriter.Workbook(filename)
    b=a.add_worksheet()
    b.write("A1","I_out(nA)")
    b.write("B1","toxvalues")
    j=67
    for i in range(18):
        if i<9:
            h='vthen'+str(i+1)+'n'
            x=chr(j)+'1'
            b.write(x,h)
            j+=1 
        else:
            h='vthep'+str(i-8)+'p'
            x=chr(j)+'1'
            b.write(x,h)
            j+=1
    for i in range(n):
        b.write(i+1,0,current[i])
        b.write(i+1,1,toxvalues[i])
    k=2
    for arr in vtharray:
        for i in range(n):
            b.write(i+1,k,arr[i])
        k+=1
    a.close()

n=int(input('1:for tox variation \n2:for vth variation \n3:for tox and vth(both) \n4:for channel length \nplease enter a number(1 or 2 or 3 or 4)\n'))

if n!=1 and n!=2 and n!=3 and n!=4:
    print('enter a valid number')
if (n==1):
    sigma=float(input('Please enter the sigma value in percentage\n(for ex:for 3%,5%,7%... variation of vth just enter 3,5,7,..respectively )\n'))
    txtfile=input('Please enter the file in proper format including extension(.cir or .txt)\n')
    n1=int(input('Please enter number of variations:(How many values of tox you want to vary)\n'))
    filename=input('Please enter the csv file name\n')
    path=pathmodification(txtfile)
    toxvalues=m.toxvariation(sigma,txtfile,n1,path)
    d1=m.toxdisplay(path)
    print(d1)
    print('These are the variables to be plotted')
    a=int(input('enter the choice of x index you want to plot:'))
    b=int(input('enter the choice of y index you want to plot:'))
    l=m.toxplot(a,b,n1,path)
    for i in l:
        print(i)
    m,k,s =m.cal(l) 
    print("For Real Values : ")
    print("Mean value is: ",m)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)
    plot(l,n,n1)
    toxxl(l,toxvalues,filename,path,n1)
    df = pd.DataFrame(pd.read_excel(r'E:\PROECT_Main\LTspiceXVII\LTspiceXVII\tox_n_varition_Final.xlsx'))
    custom_SVM(df)
if(n==2):
    sigma=float(input('Please enter the sigma value in percentage\n(for ex:for 3%,5%,7%... variation of vth just enter 3,5,7,..respectively )\n'))
    txtfile=input('Please enter the file in proper format including extension(.cir or .txt)\n')
    n3=int(input('Please enter the number of sample values\n'))
    filename=input('Please enter the csv file name\n')
    path=pathmodification(txtfile)
    d1=m.vthvariationdisplay(sigma,txtfile,n3,path)
    print(d1)
    print('These are the variables to be plotted')
    n1=int(input('enter the choice of x index you want to plot:'))
    n2=int(input('enter the choice of y index you want to plot:'))
    l,vthvalues=m.vthvariation(sigma,txtfile,n1,n2,n3,path)
    print('\n')
    for i in l:
        print(i)
    m,k,s=m.cal(l) 
    print("Mean value is: ",m)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)
    plt.hist(vthvalues,bins=n3)
    plt.xlabel('vth values')
    plt.ylabel('frequency')
    plt.title('Gaussian variation of vth')
    plt.show()
    plot(l,n,n3)
    vthxl(l,vthvalues,filename,path,n3)
    df = pd.DataFrame(pd.read_excel(r'E:\PROECT_Main\LTspiceXVII\LTspiceXVII\VTH_n_varition_Final.xlsx'))
    X = df.drop('I_out(nA)', axis=1)
    y = df['I_out(nA)']
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2) #random_state=42
    scaler = StandardScaler()
    X_train = scaler.fit_transform(X_train)
    X_test = scaler.fit_transform(X_test)
    svr = SVR(C=0.1, gamma=0.01, kernel='rbf')
    svr.fit(X_train, y_train)
    y_pred = svr.predict(X_test)
    c=df['I_out(nA)']
    res=[]
    for mem in y_test:
        for i,mem1 in enumerate(c):
            if mem==mem1:
                res.append(df['vthen1n'][i])
    df = pd.DataFrame({'vthen1n':res, 'Real Values': y_test, 'Predicted Values':y_pred})
    print(df.to_string())
    mean=sum(y_pred)/len(y_pred)
    s=skew(y_pred)
    k=kurtosis(y_pred)
    print("For Predicted Values : ")
    print("Mean value is: ",mean)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)
    mse = mean_squared_error(y_test, y_pred)
    print("Mean Squared Error:", mse)
if (n==3):
    sigma1=float(input('Please enter the sigma value for tox in percentage\n(for ex:for 3%,5%,7%... variation of vth just enter 3,5,7,..respectively )\n'))
    sigma2=float(input('Please enter the sigma value for vth in percentage\n(for ex:for 3%,5%,7%... variation of vth just enter 3,5,7,..respectively )\n'))
    txtfile=input('Please enter the file in proper format including extension(.cir or .txt)\n')
    n3=int(input('Please enter number of variations:(How many values of tox and vth you want to vary)\n'))
    filename=input('Please enter the csv file name\n')
    path=pathmodification(txtfile)
    d1=m.vthandtoxdisplayforuser(txtfile,sigma1,sigma2,n3,path)
    print(d1)
    print('These are the variables to be plotted')
    n1=int(input('Plese enter the choice of x index you want to plot:'))
    n2=int(input('Please enter the choice of y index you want to plot:'))
    l,vthvalues,toxvalues=m.variationforboth(txtfile,sigma1,sigma2,n1,n2,n3,path)
    print('\n')
    for i in l:
        print(i)
    m,k,s=m.cal(l) 
    print("Mean value is: ",m)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)
    plt.hist(toxvalues,bins=n3)
    plt.xlabel('tox values')
    plt.ylabel('frequency')
    plt.title('Gaussian variation of tox')
    plt.show()
    plt.hist(vthvalues,bins=n3)
    plt.xlabel('vth values')
    plt.ylabel('frequency')
    plt.title('Gaussian variation of vth')
    plt.show()
    plot(l,n,n3)
    vthandtoxxl(l,vthvalues,toxvalues,filename,path,n3)
    df = pd.DataFrame(pd.read_excel(r'E:\PROECT_Main\LTspiceXVII\LTspiceXVII\tox_n_varition_Final.xlsx'))
    X = df.drop('I_out(nA)', axis=1)
    y = df['I_out(nA)']
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
    scaler = StandardScaler()
    X_train = scaler.fit_transform(X_train)
    X_test = scaler.fit_transform(X_test)
    svr = SVR(C=0.1, gamma=0.01, kernel='rbf')
    svr.fit(X_train, y_train)
    y_pred = svr.predict(X_test)
    c=df['I_out(nA)']
    res1=[]
    res=[]
    for mem in y_test:
        for i,mem1 in enumerate(c):
            if mem==mem1:
                res1.append(df['vthen1n'][i])
                res.append(df['toxvalues'][i])
    df = pd.DataFrame({'Tox':res,'vthen1n':res1, 'Real Values': y_test, 'Predicted Values':y_pred})
    print(df.to_string())

    mean=sum(y_pred)/len(y_pred)
    s=skew(y_pred)
    k=kurtosis(y_pred)
    print("For Predicted Values : ")
    print("Mean value is: ",mean)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)
    mse = mean_squared_error(y_test, y_pred)
    print("Mean Squared Error:", mse)
if(n==4):
    print("0: v1, 1: V(n002), 2: V(n001), 3: V(vdd), 4: V(n003), 5: V(vx), 6: V(n004), 7: V(n005), 8: V(n006), 9: Id(M11), 10: Ig(M11), 11: Ib(M11), 12: Is(M11), 13: Id(M10), 14: Ig(M10), 15: Ib(M10), 16: Is(M10), 17: Id(M9), 18: Ig(M9), 19: Ib(M9), 20: Is(M9), 21: Id(M7), 22: Ig(M7), 23: Ib(M7), 24: Is(M7), 25: Id(M8), 26: Ig(M8), 27: Ib(M8), 28: Is(M8), 29: Id(M6), 30: Ig(M6), 31: Ib(M6), 32: Is(M6), 33: Id(M5), 34: Ig(M5), 35: Ib(M5), 36: Is(M5), 37: Id(M4), 38: Ig(M4), 39: Ib(M4), 40: Is(M4), 41: Id(M3), 42: Ig(M3), 43: Ib(M3), 44: Is(M3), 45: Id(M2), 46: Ig(M2), 47: Ib(M2), 48: Is(M2), 49: Id(M1), 50: Ig(M1), 51: Ib(M1), 52: Is(M1), 53: I(V1)")
    print('These are the variables to be plotted')
    n1=int(input('Plese enter the choice of x index you want to plot:'))
    n2=int(input('Please enter the choice of y index you want to plot:'))
    res = chplot(n1,n2) 
    df = pd.DataFrame(pd.read_excel(r'E:\PROECT_Main\LTspiceXVII\LTspiceXVII\VTH_P_varition_Final.xlsx'))
    m,k,s =m.cal(res) 
    print("For Real Values : ")
    print("Mean value is: ",m)
    print("kurtosis value is: ",k)
    print("skewness value is: ",s)
    X = df.iloc[:, 1].values
    y = df.iloc[:, -1].values

    from sklearn.preprocessing import StandardScaler
    sc_X = StandardScaler()
    sc_y = StandardScaler()
    X = sc_X.fit_transform(X.reshape(-1, 1))
    y = sc_y.fit_transform(y.reshape(-1, 1))

    from sklearn.model_selection import train_test_split
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

    from sklearn.svm import SVR
    regressor = SVR(C=0.1, gamma=0.01, kernel='rbf')
    regressor.fit(X_train.reshape(-1, 1), y_train.reshape(-1, 1))
    y_pred = regressor.predict(X_test.reshape(-1, 1))
    y_pred = sc_y.inverse_transform(y_pred.reshape(-1, 1))  
    z=sc_y.inverse_transform(y_test.reshape(-1, 1)).reshape(-1)
    I_values = df['I(A)']
    Vdd_values = []
    for i in range(len(z)):
        for j in range(len(I_values)):
            if z[i] == I_values[j]:
                Vdd_values.append(str(df['Channel Length'][j]))
                break
        else:
            Vdd_values.a
    df = pd.DataFrame({'Channel Length':Vdd_values, 'Real Values': sc_y.inverse_transform(y_test.reshape(-1, 1)).reshape(-1), 'Predicted Values': y_pred.reshape(-1)})
    print(df.to_string())
    mean = sum(y_pred) / len(y_pred)
    s = skew(y_pred)
    k = kurtosis(y_pred)
    print("For Predicted Values : ")
    print("Mean value is: ", mean)
    print("kurtosis value is: ", k)
    print("skewness value is: ", s)
    print('Mean Absolute Error is : ')
    print(metrics.mean_absolute_error(y_test, y_pred))
    print('Mean Squared Error is : ')
    print(metrics.mean_squared_error(y_test, y_pred))
    print('Root Mean Squared Error is :')
    print(np.sqrt(metrics.mean_squared_error(y_test, y_pred)))




